crates/biome_js_analyze/src/lint/nursery/use_guard_for_in.rs

```diff
@@ -0,0 +1,128 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsStatement, JsForInStatement};
+use biome_rowan::{AstNode, AstNodeList};
+
+declare_lint_rule! {
+    ///
+    /// Require `for-in` loops to include an `if` statement
+    ///
```

Comment:
Let's add more documentation, please. Let's explain why an `if` statement is required by this rule.

Reply:
ok

Reply:
Add a more detailed desc here